tests/infra/ccf.py

```diff
@@ -451,7 +451,7 @@ def wait_for_all_nodes_to_catch_up(self, primary, timeout=3):
                         and resp.result["term"] == term_leader
                     ):
                         joined_nodes += 1
-            if joined_nodes == self.get_running_nodes():
+            if joined_nodes == len(self.get_running_nodes()):
```

Comment:
I also noticed this minor issue which prevented early loop termination where all nodes had caught up.